prime-router/src/test/kotlin/serializers/CsvSerializerTests.kt

```diff
@@ -594,4 +594,47 @@ class CsvSerializerTests {
         }
         assertThat(err.details).isNotEmpty()
     }
+
+    @Test
+    fun `test schema changes do not affect reading`() {
```

Comment:
My understanding from the ticket was that the problem happens when a schema changes and there are messages that were for the 'old' schema still stuck in the pipeline. This does not appear to test that.

Reply:
For a unit test, it does.  The problem really happens when the internal CSV file has a number of columns that is different that the number of fields in its schema.  The exception from the logs shows where the problem is.

Reply:
Can you help me understand this?  To me this unit test looks like it is using the same schema for both Actions of the test, and simply changing the input value to have missing columns. I thought the issue was when the input remains the same but the schema itself changes